src/geometry.py

```diff
@@ -7,13 +7,19 @@
 
 
 class Geometry(QObject):
-    def __init__(self, guid = None):
+    def __init__(self, guid=None):
         super().__init__()
         self._guid = guid
         if not self._guid:
             self._guid = uuid.uuid4()
 
         self._mesh = om.TriMesh()
+        self.subdivboxtree = 0
```

Comment:
I must advise against placing member variable of SubDivBoxTree directly into `Geometry` object. 
It should be considered a class that operates on `Geometry` (or it's mesh), and not  member of the `Geometry`.
Would you please consider changing it?

Reply:
I understand your concerns about this. But do you have a recommendation for a better place of SubDivBoxTree? Pero and I could not figure a better place.

Reply:
I really do not have an answer for this. 
what i can suggest is to place it into the object that uses subdivboxtree. Otherwise, Geometry class will become cluttered with specific structure that is used by one type of objects. Using other words,  I 'd suggest to consider using https://en.wikipedia.org/wiki/Visitor_pattern

Reply:
The object that uses SubDivBoxTree is defaultSelector, which is used in GlWin. The magic about SubDivBoxTree occurs by creating the Tree when the geometry is loaded, leading to a faster selection time. So it is mandatory to save the created SubDivBoxTrees somewhere. One possibility would be to create a SubDixBoxTrees list, holding all SubDivBoxTree objects, but this would be placed either in application.py or GlWin. Pero told me I am neither allowed to touch GlWin nor application.py. 

The best approach I have is to save the created SubDivBoxTree objects in defaultSelector, but for this, defaultSelector would have to be a member of GlWin.

Sorry If I am missing something here, but considering the parts of the code I am allowed to modify, I still don't see a better place for SubDivBoxTree. I am open for any recommendation you have.

Reply:
> The best approach I have is to save the created SubDivBoxTree objects in defaultSelector, but for this, defaultSelector would have to be a member of GlWin.

Agreed!
I'll make neccesarry changes to glwin this evening, and if it is ok with you, commit to your local branch.
is it ok with you?

Reply:
Sure, definitely. 

I will wait for you and make the changes to DefaultSelector and Geometry tomorrow.

Reply:
I've done it. Pls check if everything is working correctly on your side (don't forget to execute `git pull` to update your local branch before)

Reply:
Great. I have adapted defaultSelector and geometry to the new behaviour and it works.
There is just one thing left: the SubDivBoxTree creation needs to be triggered when a geometry is added. For testing I managed this by adaption of onGeometryAdded in glwin.py:
```
@Slot()
def onGeometryAdded(self, geometry):
    self._bb =  self._bb + geometry.bbox
    for p in self.glPainters:
        p.addGeometry(geometry)
    self.update()
    self.selector.addGeometry(geometry)
```
Alternatively, an individual `@Slot()` function could be created for `self.selector.addGeometry` and added to `Signals.get().geometryAdded`. I hope one of these ways is ok for you or maybe you have a certain way in mind?

Reply:
Thanks for checking!
Regarding connecting selector to the geometry addition, please add follow the last approach you mentioned: add new member function (`onAddGeometry`) to the `DefaultSelector` and connect it directly to the signal. 

Part within GlWin you are referring to is very close to hacking and it will be rewritten.  In other words, all painters will be connected to the signal exactly as you will connect `DefaultSelector`.

Reply:
Okay, perfect. I followed your suggestion and commited the changes.